app/services/data_migrations/remove_surplus_reference_selections.rb

```diff
@@ -0,0 +1,30 @@
+module DataMigrations
+  class RemoveSurplusReferenceSelections
+    TIMESTAMP = 20210616160559
+    MANUAL_RUN = false
+
+    def change
+      unsubmitted_apps = retrieve_apps_with_surplus_reference_selections.select do |app|
+        app.application_choices.map(&:status).all? { |s| s == 'unsubmitted' }
```

Comment:
Maybe use `app.application_choices.all?(&:unsubmitted?)` or even `app.application_choices.none?(&:submitted?)` (the later depends on `submitted_at` though).